fe1-web/__tests__/model/network/FromJson.ts

```diff
@@ -0,0 +1,496 @@
+    object: ObjectType.LAO,
+    action: ActionType.CREATE,
+    id: mockLaoId,
+    name: name,
+    creation: time,
+    organizer: org,
+    witnesses: [],
+  };
+
+  const sampleUpdateLao: Partial<UpdateLao> = {
+    object: ObjectType.LAO,
+    action: ActionType.UPDATE_PROPERTIES,
+    id: Hash.fromStringArray(org.toString(), time.toString(), name),
+    name: name,
+    last_modified: CLOSE_TIMESTAMP,
+    witnesses: [sampleKey1, sampleKey2],
+  };
+
+  const sampleStateLao: Partial<StateLao> = {
+    object: ObjectType.LAO,
+    action: ActionType.STATE,
+    id: Hash.fromStringArray(org.toString(), time.toString(), name),
+    name: name,
+    creation: time,
+    last_modified: CLOSE_TIMESTAMP,
+    organizer: mockPublicKey,
+    witnesses: [sampleKey1, sampleKey2],
+    modification_id: Hash.fromStringArray(mockMessageId.toString()),
+    modification_signatures: [],
+  };
+
+  const sampleCreateMeeting: Partial<CreateMeeting> = {
+    object: ObjectType.MEETING,
+    action: ActionType.CREATE,
+    id: Hash.fromStringArray(mockLaoId.toString(), time.toString(), name),
+    name: name,
+    creation: time,
+    location: 'Lausanne',
+    start: time,
+    end: FUTURE_TIMESTAMP,
+    extra: { extra: 'extra info' },
+  };
+
+  const sampleStateMeeting: Partial<StateMeeting> = {
+    object: ObjectType.MEETING,
+    action: ActionType.STATE,
+    id: Hash.fromStringArray(mockLaoId.toString(), time.toString(), name),
+    name: name,
+    creation: time,
+    last_modified: time,
+    location: 'Lausanne',
+    start: CLOSE_TIMESTAMP,
+    end: FUTURE_TIMESTAMP,
+    extra: { extra: 'extra info' },
+    modification_id: Hash.fromStringArray(mockMessageId.toString()),
+    modification_signatures: [],
+  };
+
+  const rollCallId: Hash = Hash.fromStringArray('R', mockLaoId.toString(), time.toString(), name.toString());
+  const sampleCreateRollCall: Partial<CreateRollCall> = {
+    object: ObjectType.ROLL_CALL,
+    action: ActionType.CREATE,
+    id: rollCallId,
+    name: name,
+    creation: STANDARD_TIMESTAMP,
+    start: time,
+    location: 'Lausanne',
+    roll_call_description: 'description du rc',
+  };
+
+  const sampleOpenRollCall: Partial<OpenRollCall> = {
+    object: ObjectType.ROLL_CALL,
+    action: ActionType.OPEN,
+    id: rollCallId,
+    start: time,
+  };
+
+  const sampleReopenRollCall: Partial<OpenRollCall> = {
+    object: ObjectType.ROLL_CALL,
+    action: ActionType.REOPEN,
+    id: rollCallId,
+    start: time,
+  };
+
+  const sampleCloseRollCall: Partial<CloseRollCall> = {
+    object: ObjectType.ROLL_CALL,
+    action: ActionType.CLOSE,
+    id: rollCallId,
+    start: time,
+    end: FUTURE_TIMESTAMP,
+    attendees: [],
+  };
+
+  const sampleWitnessMessage: Partial<WitnessMessage> = {
+    object: ObjectType.MESSAGE,
+    action: ActionType.WITNESS,
+    message_id: mockMessageId,
+    signature: mockSecretKey.sign(mockMessageId),
+  };
+
+
+
+  it('should successfully create the objects', function () {
```

Comment:
a couple of general comments. 

1. If this test fails in the continuous integration pipeline, you'll have no idea why.
Ideally, you want to make small tests that verify just one tiny bit of functionality.

Compare the following two outputs
```
"... should successfully create the objects" - FAILED
```

vs

```
"... should successfully create the objects - simple lao" - SUCCESS
"... should successfully create the objects - update lao" - SUCCESS
"... should successfully create the objects - state lao" - SUCCESS
"... should successfully create the objects - create meeting" - SUCCESS
"... should successfully create the objects - update meeting" - FAILED
"... should successfully create the objects - state meeting" - SUCCESS
...
```

2. checking `.toThrow(ProtocolError)` may not (always) be sufficient to know precisely what failed.
You might be throwing errors for the wrong reason. Could we at least check that the message contains the name of the invalid parameter?

Reply:
1. I agree that is it way more clear (even though, from what I saw, Jest shows exactly where the error occured)
2. done

Reply:
Re: 1, it does show where the error occurred, but that already entails "investigating" in a way.
You should aim for the *unit* tests in your CI/CD pipeline to "scream" at you what you broke, with minimal investigation.
It's different with integration tests, but making them as screaming-clear as possible is always a good idea :)